framework/src/base/ComputeJacobianThread.C

```diff
@@ -222,6 +238,46 @@ ComputeJacobianThread::onInternalSide(const Elem *elem, unsigned int side)
 }
 
 void
+ComputeJacobianThread::onInterface(const Elem *elem, unsigned int side, BoundaryID bnd_id)
+{
+  // Pointer to the neighbor we are currently working on.
+  const Elem * neighbor = elem->neighbor(side);
+
+  // Get the global id of the element and the neighbor
+  const dof_id_type
+    elem_id = elem->id(),
+    neighbor_id = neighbor->id();
+
+  if ((neighbor->active() && (neighbor->level() == elem->level()) && (elem_id < neighbor_id)) || (neighbor->level() < elem->level()))
```

Comment:
It hasn't been a focus of mine, but I suppose that I would like the code to be as general as possible, thus I would like for it to work with adaptivity. I basically took those checks verbatim from the onInternalSide method. By cursory inspection they look like they're taking care of adaptivity, but I could definitely be wrong.